### PR TITLE
Rpc, Java Api modifications + Sync-only mode (restricts outbound broadcasts)

### PR DESCRIPTION
**Pull request by ali-sharif**

### CODE REVIEW

File: modApiServer/src/org/aion/api/server/pb/ApiAion0.java [hunk 1]
```diff
@@ -1284,6 +1340,149 @@ public ApiAion0(IAionChain ac) {
...
+                if (blkEnd - blkStart > 1000) {
+                    blkStart = blkEnd - 1000 + 1;
+
+                    if (blkStart < 0) blkStart = 0;
+                }
+
+                if (LOG.isDebugEnabled()) LOG.debug("BlockSqlByRange: range " + blkStart + "-" + blkEnd);
+
+                Long lastBlockTimestamp = null;
+                long listLength = blkEnd - blkStart + 1;
```
Comment (qoire) [inline modApiServer/src/org/aion/api/server/pb/ApiAion0.java]: Just to be clear here, this implies our domain is [blkStart, blkEnd]

File: modApiServer/src/org/aion/api/server/http/ApiWeb3Aion.java [hunk 2]
```diff
@@ -1592,6 +1599,166 @@ public RpcMsg ops_getChainHeadViewBestBlock() {
...
+
+        AionTxInfo txInfo = this.ac.getAionHub().getBlockchain().getTransactionInfo(txHash);
+
+        if (txInfo == null) return new RpcMsg(JSONObject.NULL);
+
+        AionBlock block = this.ac.getAionHub().getBlockchain().getBlockByHash(txInfo.getBlockHash());
+
+        if (block == null) return new RpcMsg(JSONObject.NULL);
+
+        AionTransaction tx = block.getTransactionsList().get(txInfo.getIndex());
```
[the remark this answers is not in the document]
Comment (ali-sharif) [reply modApiServer/src/org/aion/api/server/http/ApiWeb3Aion.java]: no reason. both cost roughly the same. changed it to get transaction from receipt.

File: modApiServer/src/org/aion/api/server/http/ApiWeb3Aion.java [hunk 3]
```diff
@@ -1592,6 +1599,166 @@ public RpcMsg ops_getChainHeadViewBestBlock() {
...
+        AionBlock mainBlock = this.ac.getBlockchain().getBlockByNumber(block.getNumber());
+        if (mainBlock == null) return new RpcMsg(JSONObject.NULL);
+
+        if (!FastByteComparisons.equal(block.getHash(), mainBlock.getHash())) {
+            return new RpcMsg(JSONObject.NULL);
+        }
+
+        // ok so now we have a mainchain block
+
+        BigInteger blkReward = ((AionBlockchainImpl)ac.getBlockchain()).getChainConfiguration()
```
[the remark this answers is not in the document]
Comment (ali-sharif) [reply modApiServer/src/org/aion/api/server/http/ApiWeb3Aion.java]: yes. understood. unfortunately, we don't put that number in the block summary (those fee rewards are applied directly in `TransactionExecutor.execute()`).

File: modApiServer/src/org/aion/api/server/pb/ApiAion0.java (hunk 1 above)
Comment (ali-sharif) [reply modApiServer/src/org/aion/api/server/pb/ApiAion0.java]: yes, domain is [blkStart, blkEnd]

File: modApiServer/src/org/aion/api/server/http/ApiWeb3Aion.java [hunk 4]
```diff
@@ -688,6 +689,12 @@ else if (_params instanceof JSONObject) {
         byte[] hash = ByteUtil.hexStringToBytes(_hash);
         AionBlock block = this.ac.getBlockchain().getBlockByHash(hash);
 
+        AionBlock mainBlock = this.ac.getAionHub().getBlockchain().getBlockByNumber(block.getNumber());
+        if (!FastByteComparisons.equal(block.getHash(), mainBlock.getHash())) {
```
[the remark this answers is not in the document]
Comment (qoire) [reply modApiServer/src/org/aion/api/server/http/ApiWeb3Aion.java]: LGTM

File: modApiServer/src/org/aion/api/server/http/ApiWeb3Aion.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (qoire) [reply modApiServer/src/org/aion/api/server/http/ApiWeb3Aion.java]: Understood, perhaps we can raise this as a future enhancement

File: modApiServer/src/org/aion/api/server/pb/ApiAion0.java (hunk 1 above)
Comment (qoire) [reply modApiServer/src/org/aion/api/server/pb/ApiAion0.java]: LGTM

**Pull request merged**